build.go

```diff
@@ -0,0 +1,111 @@
+package composer
+
+import (
+	"github.com/paketo-buildpacks/packit/v2"
+	"github.com/paketo-buildpacks/packit/v2/chronos"
+	"github.com/paketo-buildpacks/packit/v2/postal"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+// Note that Go 1.18 requires faux 0.21.0 (https://github.com/ryanmoran/faux/releases/tag/v0.21.0)
+
+//go:generate faux --interface DependencyManager --output fakes/dependency_manager.go
+type DependencyManager interface {
+	Resolve(path, id, version, stack string) (postal.Dependency, error)
+	Deliver(dependency postal.Dependency, cnbPath, layerPath, platformPath string) error
+}
+
+//go:generate faux --interface EntryResolver --output fakes/entry_resolver.go
+type EntryResolver interface {
+	Resolve(name string, entries []packit.BuildpackPlanEntry, priorites []interface{}) (packit.BuildpackPlanEntry, []packit.BuildpackPlanEntry)
+	MergeLayerTypes(string, []packit.BuildpackPlanEntry) (launch, build bool)
+}
+
+func Build(
+	logger LogEmitter,
+	dependencyManager DependencyManager,
+	entryResolver EntryResolver,
+	clock chronos.Clock) packit.BuildFunc {
+	return func(context packit.BuildContext) (packit.BuildResult, error) {
+		logger.Title("%s %s", context.BuildpackInfo.Name, context.BuildpackInfo.Version)
+		logger.Process("Resolving Composer version")
+
+		priorities := []interface{}{
+			"BP_COMPOSER_VERSION",
+			"",
```

Comment:
Why include `""`?

Reply:
I assumed that meant default but looks unnecessary